osu.Game/Screens/Play/HUD/HealthDisplay.cs

```diff
@@ -3,15 +3,30 @@
 
 using osu.Framework.Bindables;
 using osu.Framework.Graphics.Containers;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
 
 namespace osu.Game.Screens.Play.HUD
 {
+    /// <summary>
+    /// A container for components displaying the current player health.
+    /// Gets bound automatically to the <see cref="HealthProcessor"/> when inserted to <see cref="DrawableRuleset.Overlays"/> hierarchy.
+    /// </summary>
     public abstract class HealthDisplay : Container
     {
         public readonly BindableDouble Current = new BindableDouble
         {
             MinValue = 0,
             MaxValue = 1
         };
+
+        /// <summary>
+        /// Bind the tracked fields of <see cref="HealthProcessor"/> to this health display.
+        /// </summary>
+        /// <param name="processor"></param>
+        public virtual void BindHealthProcessor(HealthProcessor processor)
```

Comment:
This again seems to be prematurely engineered, why is this `virtual`?

Reply:
This was marked virtual to allow implementors of the `HealthDisplay` class to bind to other properties of the `HealthProcessor` if they need to but I guess this can be reverted to a normal method in a lack of a current usage overriding the method.

Reply:
As a general rule, make things as simple as they can be when going forward with something new. Don't try to anticipate future usages, most of the time you'll fail anyway. Things can be changed when they need to be changed.